pages/api/configuration.ts

```diff
@@ -5,83 +5,127 @@ import { domainMiddleware, jwtVerifyMiddleware } from "../../lib/middlewares";
 import MiddlewareError from "../../utils/MiddlewareError";
 import { getAuthToken } from "../../lib/environment";
 import {
-  FetchAppMetadataDocument,
-  FetchAppMetadataQuery,
   UpdateAppMetadataDocument,
   UpdateAppMetadataMutation,
-  MetadataItem,
-  MetadataInput,
+  FetchAppMetafieldsQuery,
+  FetchAppMetafieldsDocument,
+  FetchAppMetafieldsQueryVariables,
+  UpdateAppMetadataMutationVariables,
 } from "../../generated/graphql";
-
-const CONFIGURATION_KEYS = [
-  "THAT_SECRET",
-  "THIS_VALUE",
-];
-
-const prepareMetadataFromRequest = (input: MetadataInput[] | MetadataItem[]) =>
-  input
-    .filter(({ key }) => CONFIGURATION_KEYS.includes(key))
-    .map(({ key, value }) => ({ key, value }));
-
-const prepareResponseFromMetadata = (input: MetadataItem[]) => {
-  const output: MetadataInput[] = [];
-  for (const configurationKey of CONFIGURATION_KEYS) {
-    output.push(
-      input.find(({ key }) => key === configurationKey) ?? { key: configurationKey, value: "" }
-    );
-  }
-  return output.map(({ key, value }) => ({ key, value }));
-};
+import {
+  prepareMetadataFromRequest,
+  prepareResponseFromMetadata,
+} from "@/backend/configuration";
 
 const handler: NextApiHandler = async (request, response) => {
```

Comment:
So this function is very long and it's hard to track what's going on, could we maybe split into smaller functions?

Reply:
Yes, I also think so, I would like to refactor api in another pull request.